A28-xds-traffic-splitting-and-routing.md

```diff
@@ -0,0 +1,1095 @@
+policy will be picked using the same algorithm as [service
+config](https://github.com/grpc/grpc-proto/blob/master/grpc/service_config/service_config.proto#L297).
+
+*   The first known (registered) policy name in the list will be picked
+    *   If none is registered, the config is invalid
+*   The config for the picked policy will be validated
+    *   If the validation fails, the config is invalid. We won’t move on to the
+        next entry.
+
+Action names are used as ID for the child policies. If an action name exists
+in the previous service config, the action’s balancer config will be sent as
+an update to the corresponding child policy (instead of creating a new child
+policy for it).
+
+NOTE: this LB policy is named `xds_routing` (instead of just routing) because
+we plan to make it deeply integrated with the xds resolver as part of
+RouteAction design.
+
+### Picking
+
+When making a pick, the first match wins.
+
+For example, with config
+
+```
+{prefix: "/MyService"}:             childPolicy-1
+{path:   "/MyService/MyMethod"}:    childPolicy-2
+```
+
+`ChildPolicy-1` will be picked for `/MyService/MyMethod`, even though the
+second route is an exact match. It is this way because Envoy always picks the
+first matched route. And with header matching, finding the exact match can be
+tricky.
+
+The pick will always be delegated to the picker from the matched child
+policy, even if it’s not in connectivity state READY (as opposed to
+weighted\_target will only pick the READY child policies).
+
+There will also be cases where for an RPC, a match is found, but the child
+policy hasn’t generated the picker (for example, the child policy is still
+initializing, and no picker has been created). When this happens, the RPCs
+will be always queued until a picker is updated.
+
+If no match can be found, the RPC will fail with code Unavailable, and
+detailed information in the error message.
+
+When the policy gets an update, including:
+
+*   Balancer config update from the resolver (route update)
+*   Child policy picker update
+
+The queued RPCs will all re-pick, even though it could get the same result
+because its route isn’t updated, or it’s matching child policy hasn’t
+changed. (This is less than ideal, and some of the picks are wasted. But this
+is necessary because the RPCs are queued by the channel, and the LB policy
+cannot only re-pick some of them.)
+
+The connectivity state of the policy will be the aggregated state of all the
+child policies. Even though this is not true for a specific RPC (the actual
+state is the state of the child policy that will be picked), it represents
+the overall state of the channel.
+
+## xDS resolver and xDS client
+
+The xDS resolver’s overall behavior will be similar as before. It will call
+the xDS client, and will send a service config generated from
+RouteConfiguration (the RDS response) to the parent gRPC client channel.
+
+The xDS client will make an LDS request, with resource\_names set to the
```

Comment:
Do we need to repeat stuff that is not new or has not changed (the one at https://github.com/grpc/proposal/blob/master/A27-xds-global-load-balancing.md#xds-resolver)

Reply:
Removed